lib/alchemy/config_loader.rb

```diff
@@ -0,0 +1,69 @@
+module Alchemy
+  class ConfigLoader
+    attr_reader :which
+
+    def initialize(which)
+      @which = which.sub(/\.yml$/, '')
+    end
+
+    # Reads the config file named +<which>.yml+ from +config/alchemy/+ folder.
+    #
+    def load_all
+      if paths.empty?
+        raise LoadError, "Could not find #{which}.yml file! Please run `rails generate alchemy:scaffold`"
+      else
+        # OPTIMIZE: remove duplicates?
+        paths.map(&method(:load_file)).reduce(&method(:merge))
```

Comment:
Replacing `inject` with its alias `reduce` is not what I meant by being more explicit ;)

I think we should split this up into two parts. One "loading the ymls" part.

```ruby
def loaded_files
  @_loaded_files ||= paths.map do |path|
    load_file(path)
  end
end
```

Then we need to determine a good rule set of how we merge these files in a definite order:

1. The file from the host app has highest priority
2. The file from any engine has second priority (no need to provide a ordering inside these)